libnest2d_js/libnest2d_js.cpp

```diff
@@ -0,0 +1,247 @@
+#ifndef LIBNEST2D_JS_H
+#define LIBNEST2D_JS_H
+//Copyright (c) 2022 Ultimaker B.V.
+//libnest2d_js is released under the terms of the LGPLv3 or higher.
+
+// Emscripten Embind bindings for libnest2d
+#include <libnest2d/libnest2d.hpp>
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+#include <libnest2d/backends/clipper/geometries.hpp>
+#include <libnest2d/placers/bottomleftplacer.hpp>
+
+using namespace emscripten;
+using namespace libnest2d;
+using namespace placers;
+
+// Type aliases to match Python bindings
+using Point = PointImpl;
+using Box = _Box<Point>;
+using Circle = _Circle<Point>;
+using Item = _Item<PolygonImpl>;
+using NfpConfig = NfpPConfig<PolygonImpl>;
+using Polygon = PolygonImpl;
+
+// Add aliases for angle types
+using Radians = libnest2d::Radians;
+using Degrees = libnest2d::Degrees;
+
+// Declare value types for TypeScript generation
+EMSCRIPTEN_DECLARE_VAL_TYPE(PointList);
+EMSCRIPTEN_DECLARE_VAL_TYPE(ItemList);
+EMSCRIPTEN_DECLARE_VAL_TYPE(DoubleList);
+EMSCRIPTEN_DECLARE_VAL_TYPE(ResultAndItem);
+
+void testFunction() {
+    std::vector<Item> input;
+
+    auto volume = libnest2d::Box(1000, 1000);
+
+    std::vector<Item> items;
+
+    items.emplace_back(libnest2d::Item({
+                                               Point(5, 10),
+                                               Point(10, 10),
+                                               Point(0, 0)
+                                       }));
+    auto& long_thin_triangle = items.back();
+
+    items.emplace_back(libnest2d::Item({
+                                               Point(0, 10),
+                                               Point(10, 10),
+                                               Point(10, 0),
+                                               Point(0, 0),
+                                       }));
+    auto& square = items.back();
+
+    items.emplace_back(libnest2d::Item({
+                                               Point(5, 10),
+                                               Point(10, 0),
+                                               Point(0, 0)
+                                       }));
+    auto& equilateral_triangle = items.back();
+
+    auto num_bins = libnest2d::nest(items, volume);
+    std::cout << "Number of bins used: " << num_bins << std::endl;
+}
+
+// Helper function to convert a Point to a JavaScript object
+emscripten::val pointToJSObject(const Point& point) {
+    emscripten::val obj = emscripten::val::object();
+    obj.set("x", long{ static_cast<long>(getX(point)) } );
+    obj.set("y", long{ static_cast<long>(getY(point)) } );
+    return obj;
+}
```

Comment:
What's the benifit of this function if we already have a `Point` class exposed? Why not return that one? It saves copying a point data since if you return a `Point` it can keep the constructor in the WASM environment and only return a reference to javascript.